opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -53,57 +47,56 @@ def collect(self) -> Optional[_PointVarT]:
-):
-    def __init__(self, instrument_is_monotonic: bool):
-        super().__init__(instrument_is_monotonic)
-        self._value = 0
+class SumAggregation(Aggregation[Sum]):
+    def __init__(
+        self,
+        instrument_is_monotonic: bool,
+        instrument_temporality: AggregationTemporality,
+    ):
+        super().__init__()
+        self._instrument_is_monotonic = instrument_is_monotonic
+
+        if instrument_temporality is AggregationTemporality.DELTA:
+            self._value = 0
+
+        else:
+            self._value = None
+
         self._start_time_unix_nano = _time_ns()
+        self._instrument_temporality = instrument_temporality
 
     def aggregate(self, measurement: Measurement) -> None:
-        with self._lock:
-            self._value = self._value + measurement.value
+
+        if self._instrument_temporality is AggregationTemporality.DELTA:
+            with self._lock:
+                self._value = self._value + measurement.value
+        else:
+            with self._lock:
+                self._value = measurement.value
 
     def collect(self) -> Optional[Sum]:
         """
         Atomically return a point for the current value of the metric and
         reset the aggregation value.
         """
-        now = _time_ns()
-
-        with self._lock:
-            value = self._value
-            start_time_unix_nano = self._start_time_unix_nano
+        if self._instrument_temporality is AggregationTemporality.DELTA:
+            now = _time_ns()
 
-            self._value = 0
-            self._start_time_unix_nano = now + 1
+            with self._lock:
+                value = self._value
+                start_time_unix_nano = self._start_time_unix_nano
 
-        return Sum(
-            aggregation_temporality=AggregationTemporality.DELTA,
-            is_monotonic=self._instrument_is_monotonic,
-            start_time_unix_nano=start_time_unix_nano,
-            time_unix_nano=now,
-            value=value,
-        )
+                self._value = 0
+                self._start_time_unix_nano = now + 1
 
+            return Sum(
+                aggregation_temporality=AggregationTemporality.DELTA,
+                is_monotonic=self._instrument_is_monotonic,
+                start_time_unix_nano=start_time_unix_nano,
+                time_unix_nano=now,
+                value=value,
+            )
 
-class AsynchronousSumAggregation(
-    _InstrumentMonotonicityAwareAggregation, Aggregation[Sum]
-):
-    def __init__(self, instrument_is_monotonic: bool):
-        super().__init__(instrument_is_monotonic)
-        self._value = None
-        self._start_time_unix_nano = _time_ns()
-
-    def aggregate(self, measurement: Measurement) -> None:
-        with self._lock:
-            self._value = measurement.value
-
-    def collect(self) -> Optional[Sum]:
-        """
-        Atomically return a point for the current value of the metric.
-        """
         if self._value is None:
```

Comment:
Shouldn't this be the first piece of logic returned?

Reply:
Fixed